lib/string.js

```diff
@@ -196,6 +198,59 @@ internals.String.prototype.email = function (isEmailOptions) {
 };
 
 
+internals.String.prototype.uri = function (uriOptions) {
+    var regex = internals.uriRegex,
+        customScheme;
+
+    if (uriOptions) {
+        Hoek.assert(typeof uriOptions === 'object', 'uri options must be an object');
+
+        if (uriOptions.scheme) {
+            Hoek.assert(uriOptions.scheme instanceof RegExp || typeof uriOptions.scheme === 'string' || Array.isArray(uriOptions.scheme), 'scheme must be a RegExp, String, or Array');
+            // If someone wants to match HTTP or HTTPS for example then we need to support both RegExp and String so we don't escape their pattern unknowingly.
+            if (uriOptions.scheme instanceof RegExp) {
```

Comment:
Sounds good. this changed the error messages a little bit, so now they will mention "at position 0" when only a single String is passed, but I think that's okay.